### ISSUE TITLE
Can not return Proper Class according to document_dir (Entity)

### ISSUE THREAD

**Issue by weburnit** [ongr-io/ElasticsearchBundle#729]
```php
public function getClassName($className)
    {
        return $this->finder->getNamespace($className);
    }
```
Somehow, we must retrieve $documentDirectory and pass through `$finder->getNamespace` in case $className does not use default `Document` as directory

**Comment by saimaz** [comment]
Seems a missed place from custom `document_dir` feature. I'll take a look.

**Comment by weburnit** [comment]
Hello @saimaz, regarding Document repository, I prefer an explicit way to get specific repository:
```php
$manager->getRepository(\App\Namespace\Document::class);
```
rather than provide a string like Doctrine style `AppNamespaceBundle:Document` which makes everything more complicated.

**Comment by saimaz** [comment]
It's not doctrine style. It's how Symfony get's resources, templates, assets, entities, etc.

**Comment by weburnit** [comment]
yeah... whatever is, providing getting Repository through class, we are able to propose this bundle into different framework.

**Comment by saimaz** [comment]
Should be fixed via #763. Can you check if it solves a problem for you @weburnit ?
